tests/test_metadata/test_consolidated.py

```diff
@@ -521,7 +521,7 @@ async def test_consolidated_metadata_v2(self):
                         dtype=dtype,
                         attributes={"key": "a"},
                         chunks=(1,),
-                        fill_value=0,
+                        fill_value=None,
```

Comment:
So to clarify, this change is a breaking one for writing v2 data? Is this something we want to avoid breaking?

Reply:
it's change to how v2 metadata is written. the actual array data will be written as before. Because the signatures of the affected functions are inconsistent, there is _no way_ to achieve the goal of this PR without changing the default behavior of some of those functions, which will result in different metadata being written to disk as an effect.

Reply:
so to your questions: it's breaking if someone relied on the default metadata document produced by some, but not all, of our array creation routines. And no, we do not want to avoid breaking, because the cost of a confusing, inconsistent API is worth paying in this case

Reply:
I think we should adjust the versioning policy, and also release the changes in this PR in a minor (non-patch) release. We were already planning on making breaking changes without incrementing the major version number -- deprecated functions like `create` were slated to be removed long before a 4.x release. So I think the versioning policy needs updating.